trackers/huggingface/graphs.py

```diff
@@ -650,26 +667,33 @@ def device_funnel(df: pd.DataFrame) -> None:
 
     # Show Sankey graph with percentages
     sk_val = {
-        "All models": f"{summ.all_models} models - 100%",
-        "Converts to ONNX": f"{summ.base_onnx} models - "
-        + str(int(100 * summ.base_onnx / summ.all_models))
-        + "%",
-        "Optimizes ONNX file": f"{summ.optimized_onnx} models - "
-        + str(int(100 * summ.optimized_onnx / summ.all_models))
-        + "%",
-        "Converts to FP16": f"{summ.fp16_onnx} models - "
-        + str(int(100 * summ.fp16_onnx / summ.all_models))
-        + "%",
-        "Acquires Nvidia Perf": f"{summ.nvidia} models - "
-        + str(int(100 * summ.nvidia / summ.all_models))
-        + "% (Nvidia)",
-        "Acquires Groq Perf": f"{summ.groq} models - "
-        + str(int(100 * summ.groq / summ.all_models))
-        + "% (Groq)",
-        "Acquires x86 Perf": f"{summ.x86} models - "
-        + str(int(100 * summ.x86 / summ.all_models))
-        + "% (x86)",
+        "All models": device_funnel_metrics(summ.all_models, summ.all_models),
+        "Converts to ONNX": device_funnel_metrics(summ.base_onnx, summ.all_models),
+        "Optimizes ONNX file": device_funnel_metrics(
+            summ.optimized_onnx, summ.all_models
+        ),
+        "Converts to FP16": device_funnel_metrics(summ.fp16_onnx, summ.all_models),
+        "Acquires Nvidia Perf": device_funnel_metrics(summ.nvidia, summ.all_models)
+        + " (Nvidia)",
+        "Acquires Groq Perf": device_funnel_metrics(summ.groq, summ.all_models)
+        + " (Groq)",
+        "Acquires x86 Perf": device_funnel_metrics(summ.x86, summ.all_models)
+        + " (x86)",
     }
+
+    # Calculate bar heights for each of the devices
+    # Bar height is proportional to the number of models benchmarked by each device
+    default_bar_size = 1
+    x86_models = max(summ.x86, 1)
+    nvidia_models = max(summ.x86, 1)
+    groq_models = max(summ.groq, 1)
+    num_models_benchmarked = x86_models + nvidia_models + groq_models
+    target_combined_height = max(default_bar_size, summ.fp16_onnx)
+    device_bar_size = target_combined_height / 3
+    x86_bar_size = x86_models * target_combined_height / num_models_benchmarked
+    nvidia_bar_size = nvidia_models * target_combined_height / num_models_benchmarked
+    groq_bar_size = groq_models * target_combined_height / num_models_benchmarked
```

Comment:
We can also significantly simplify this logic if we simply say that the height of each bar is the same (1/3 of the target_combined_height) instead of making it proportional to the number of models benchmarked by each device. Let me know what you think @jeremyfowers.

Reply:
heh i didn't even notice that is was proportional since they are all so small (very few px of height even in the biggest case). So I definitely wouldn't miss it if they all just became 1/3.

I suggest keeping in mind that we will add AMD, ARM, etc. to the graph, so the question is how will you scale to more devices? Probably by making the graph taller to accomodate all of them, not by having them keep splitting up a fixed px budget?

Reply:
Simplified logic, which will also make it more future proof and easier to read in edge cases.